File: exchange-frontend/src/app/model/transaction.ts
export interface Transaction {
    id: number;
    txid: string;
    from_currency: string;
    to_currency: string;
    email: string;
    sent_amount: number;
    received_amount: number;
    exchange_rate: number;
    exchange_fee: number;
    address: string,
    status: string;
    created_at: string;
};

File: exchange-backend/app/models/transaction.rb
# frozen_string_literal: true

class Transaction < ApplicationRecord
  validates :from_currency, presence: true
  validates :to_currency, presence: true
  validates :sent_amount, presence: true
  validates :exchange_rate, presence: true
  validates :address, presence: true
  validates :email, presence: true
end
